packages/framework/src/handler.ts

```diff
@@ -223,17 +241,27 @@ export class NovuRequestHandler<Input extends any[] = any[], Output = any> {
   private getGetActionMap(workflowId: string, stepId: string): Record<GetActionEnum, () => Promise<IActionResponse>> {
     return {
       [GetActionEnum.DISCOVER]: async () => {
-        const result = await this.client.discover();
+        const errors = this.client.getErrors();
+
+        if (errors.length !== 0) {
+          return this.createResponse(HttpStatusEnum.UNPROCESSABLE_ENTITY, errors);
+        }
+        const result = this.client.discover();
 
         return this.createResponse(HttpStatusEnum.OK, result);
       },
       [GetActionEnum.HEALTH_CHECK]: async () => {
-        const result = await this.client.healthCheck();
+        const result = this.client.healthCheck();
 
         return this.createResponse(HttpStatusEnum.OK, result);
       },
       [GetActionEnum.CODE]: async () => {
-        const result = await this.client.getCode(workflowId, stepId);
+        const errors = this.client.getErrors();
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
❓ **question:** ‏ same as execute, what is the usecase for code? I'm not sure how it gets used in the studio and the bridge api client doesn't seem to use the action.

Reply:
cc: @rifont ☝️